From: =?UTF-8?q?Valdis=20Kl=20=C4=93=20tnieks?= <valdis.kletnieks@vt.edu>
Date: Sat, 9 May 2020 00:47:19 -0400
Subject: bpfilter: document build requirements for bpfilter_umh
Patch-mainline: v5.8-rc1
Git-commit: 9f64fbdb774838db1445268fa8c46041fb1c28ab
References: bsc#1177028

It's not intuitively obvious that bpfilter_umh is a statically linked binary.
Mention the toolchain requirement in the Kconfig help, so people
have an easier time figuring out what's needed.

Signed-off-by: Valdis Kletnieks <valdis.kletnieks@vt.edu>
Signed-off-by: Masahiro Yamada <masahiroy@kernel.org>
Acked-by: Gary Lin <glin@suse.com>
---
 net/bpfilter/Kconfig |    4 ++++
 1 file changed, 4 insertions(+)

--- a/net/bpfilter/Kconfig
+++ b/net/bpfilter/Kconfig
@@ -13,4 +13,8 @@ config BPFILTER_UMH
 	default m
 	help
 	  This builds bpfilter kernel module with embedded user mode helper
+
+	  Note: your toolchain must support building static binaries, since
+	  rootfs isn't mounted at the time when __init functions are called
+	  and do_execv won't be able to find the elf interpreter.
 endif
